native/general_single.go

```diff
@@ -12,6 +12,21 @@ import (
 	math "github.com/gonum/blas/native/internal/math32"
 )
 
+func checkMatrix(m, n int, a []float32, lda int) {
```

Comment:
Rename this checkMatrix64, then you can generate the float32 version and change all the calls to be appropriate.